op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -144,15 +183,22 @@ func readAncientBlocks(ctx context.Context, freezer *rawdb.Freezer, startBlock,
 				return fmt.Errorf("failed to read tds from old freezer: %w", err)
 			}
 
+			if err := CheckRLPBlockRangeForGaps(blockRange, count); err != nil {
+				return fmt.Errorf("failed to ensure ancient block range has no gaps: %w", err)
+			}
+
```

Comment:
I think it would be neater, simpler and safer if you checked for consistency between block ranges here, instead of in  `transformBlocks`.

You could adjust the Check to accept two ranges and check the transition between them. E.g

`CheckRLPBlockRangeForGaps(prevBlockRange, currBlockRange RLPBlockRange, expectedLength uint64)`

If prev is non nil you check that the last block of prev is the parent of the first block of curr.